### PR TITLE
Templatize the scaling policy for metrics-server

### PR DESCRIPTION
**Pull request by vantuvt**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by vantuvt** [comment]
Friendly ping.

File: cluster/addons/metrics-server/metrics-server-deployment.yaml
Comment (gmarek) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml]: Impossible to change now, too bad.

**Comment by vantuvt** [comment]
That flag configures the minimum cluster that the addon resizer will scale
resources to. Previously it was a hard-coded value of 16.

File: cluster/addons/metrics-server/metrics-server-deployment.yaml
Comment (gmarek) [inline cluster/addons/metrics-server/metrics-server-deployment.yaml]: For consistency I'd templetize cpu as well - all settings will be in a single place.

File: cluster/gce/gci/configure-helper.sh
Comment (gmarek) [inline cluster/gce/gci/configure-helper.sh]: Is there a reason why you want to put a no-op logic here?

File: cluster/addons/metrics-server/metrics-server-deployment.yaml
Comment (piosz) [inline cluster/addons/metrics-server/metrics-server-deployment.yaml]: Inconsistency: lack of Mi here, Mi below.

File: cluster/gce/gci/configure-helper.sh
Comment (piosz) [inline cluster/gce/gci/configure-helper.sh]: Same as above

File: cluster/gce/gci/configure-helper.sh
Comment (piosz) [inline cluster/gce/gci/configure-helper.sh]: Same as above

File: cluster/addons/metrics-server/metrics-server-deployment.yaml
Comment (piosz) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml]: Please add a comment explain what this flag is for.

**Review by piosz — CHANGES_REQUESTED** [review]

**Comment by piosz** [comment]
@vantuvt could you please explain what will be the behavior of addon manager after this change is merged?

File: cluster/addons/metrics-server/metrics-server-deployment.yaml
Comment (vantuvt) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml]: The implementation of this change mirrors the implementation currently used for heapster. In both cases the extra-cpu flag is not being tuned. For consistency with existing code and since it is not critical to the functionality I'm looking to implement I chose not to templatize the value of that flag. I can, however, implement a follow-up change to do the proposed clean-up if desired.

File: cluster/addons/metrics-server/metrics-server-deployment.yaml
Comment (vantuvt) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml]: The implementation of this change is based on how the heapster yaml has been templatized and is consistent with that implementation. I too noticed this inconsistency, but didn't want to make things inconsistent with existing code. I can implement a follow-up PR to clean this up.

File: cluster/gce/gci/configure-helper.sh
Comment (vantuvt) [reply cluster/gce/gci/configure-helper.sh]: Can you please clarify which previous comment you're referring to?

File: cluster/gce/gci/configure-helper.sh
Comment (vantuvt) [reply cluster/gce/gci/configure-helper.sh]: Can you please clarify which previous comment you're referring to?

**Comment by vantuvt** [comment]
@piosz - This PR does not change the behaviour of addon manager in any way. This is a no-op change that merely provides the framework needed to support new scaling policy values that I will be setting in a follow-up PR. I am working with others to get final approval on the new numbers.

File: cluster/gce/gci/configure-helper.sh
Comment (gmarek) [reply cluster/gce/gci/configure-helper.sh]: Are you saying that you want to check in unnecessary code to make local testing easier, or am I understanding this wrong?

Comment (vantuvt) [reply cluster/gce/gci/configure-helper.sh]: This code is necessary to support the final implementation of the functionality being added. It is not critical to this specific PR. My goal with this PR was to have the foundation ready for a follow-up PR that would update the numbers to be used when the ENABLE_SYSTEM_ADDON_RESOURCE_OPTIMIZATIONS environment variable is set to true. Given that I am now close to getting those numbers finalized I'll probably update this PR with those values as soon as I've got approval from the appropriate stakeholders. In the meantime, I'd still like to have the rest of the changes reviewed in advance. Thanks.

**Review by piosz — APPROVED** [review]

**Pull request merged**
